awx/main/credential_plugins/hashivault.py

```diff
@@ -227,6 +227,7 @@ def method_auth(**kwargs):
     cacert = kwargs.get('cacert', None)
 
     sess = requests.Session()
+    sess.mount(url, requests.adapters.HTTPAdapter(max_retries=5))
```

Comment:
does HTTPAdapter have backoff between retries? or does it wait a fixed amount of time between posts/gets? is this configurable?

what http code triggers the retry?

Reply:
`HTTPAdapter` passes the `max_retries` argument to `urlopen` from `urllib3.poolmanager.PoolManager`. 
Based on the description at  https://urllib3.readthedocs.io/en/stable/reference/urllib3.connectionpool.html#urllib3.HTTPConnectionPool.urlopen there is no builtin back-off behaviour, unless we use a `Retry` object in which case all of this is configurable.

`max_retries` is only used for things like DNS lookup errors, connection time outs, ... as soon as the request hit the server this doesn't apply anymore.
See https://docs.python-requests.org/en/latest/api/#requests.adapters.HTTPAdapter